gokart/mypy.py

```diff
@@ -402,24 +407,115 @@ def _infer_task_on_kart_attr_init_type(self, sym: SymbolTableNode, context: Cont
 
         return default
 
+    def _generate_type_from_parameters(self, parameter: Expression) -> Optional[Type]:
+        """
+        Generate default type from Parameter.
+        For example, when parameter is `luigi.parameter.Parameter`, this method should return `str` type.
+        """
+        parameter_name = _extract_parameter_name(parameter)
+        if parameter_name is None:
+            return None
+
+        result_type: Optional[Type] = None
```

Comment:
result_type is confused a little because it does not consider optional first 
so, why don't you rename `result_type` with `underlying_type`?

```suggestion
        underlying_type: Optional[Type] = None
```

Then, plz see the following review